titus-common/src/main/java/com/netflix/titus/common/framework/simplereconciler/internal/DefaultManyReconciler.java

```diff
@@ -144,14 +148,19 @@ public DefaultManyReconciler(
                     executor.close();
                 }
             }
-        }).publishOn(notificationScheduler);
+        }).publishOn(notificationSchedulerRef.get());
     }
 
     @Override
     public Mono<Void> close() {
         return Mono.defer(() -> {
             stateRef.compareAndSet(ReconcilerState.Running, ReconcilerState.Closing);
-            return closedProcessor;
+
+            return closedProcessor.doFinally(signal -> {
+                reconcilerSchedulerRef.close();
+                // TODO There may be pending notifications, so we have to delay the actual close. There must be a better solution than timer.
```

Comment:
Is there no way to get a pending count here?

Reply:
Yes, we could do it but it would be a bit more complicated. I will fix that when working again on this code.